### ISSUE TITLE
Halyard docs for mapping roles from OIDC in OAuth2 UserInfoMapping not clear.

### ISSUE THREAD

**Issue by shubhamc183** [spinnaker/spinnaker#5938]
Recently I was trying to setup Spinnaker for authN and authZ via Keycloak, OIDC where I wanted to map roles.

I faced an issue for quite a time setting the AuthZ for quite some time where I was not able to map roles from external sources.

`~/.hal/config`
```
security:
  basic:
    enabled: true
  user: {}
  oauth2:
    enabled: true
    client:
      clientId: XXX
      clientSecret: XXX
      accessTokenUri: XXX
      userAuthorizationUri: XXX
    resource:
      userInfoUri: XXX
    userInfoMapping: {}
    provider: OTHER
```

And `.hal/default/profiles/gate-local.yml`
```
security:
  oauth2:
    userInfoMapping:
      email: email
      firstName: given_name
      lastName: family_name
      username: preferred_username
      roles: groups
    client:
      scope: openid,spinroles

```
**The important detail here is that `userInfoMapping` I have to keep in `.hal/default/profiles/gate-local.yml` and not in `~/.hal/config`. Else it'll result into an error like this https://github.com/spinnaker/spinnaker/issues/5902#issue-655753819**

I was not able to find the exact documentation which states that, I spent like 10 days on this so opening an issue to document if not present already.

https://github.com/spinnaker/spinnaker/issues/5902

**Comment by dorbin** [comment]
@spinnaker/sig-security

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by shubhamc183** [comment]
Any updates or hints?

@robzienert, @dorbin

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.

**Comment by ishallbethat** [comment]
why this issue closed ? it's not fixed yet

**Comment by VayshaYsno** [comment]
Any updates? I see an opened merge request where this is fixed, but no one cares for about 3 years now: [fix(oauth2): add roles to user info mapping](https://github.com/spinnaker/halyard/pull/1880) 

I'm now trying to deploy Spinnaker in k8s with Spinnaker-Operator, and this issue ruins my SSO and RBAC system...
I saw a solution somewhere on GitHub, that there is a way to override `userAttributeMapping` via `gate-local.yml`, guys used to do main setup in main config, and specialize `userAttributeMapping` providing `roles:` in `gate-local.yml`. 
For me, it worked wrong way: `gate-local.yml` passed to the pod, but wasn't applied. Seems like idk how to configure correct `*-local` files.
There is this case I've mentioned above: [Halyard not allowing to map roles from OIDC, KeyCloak, in OAuth2 UserInfoMapping](https://github.com/spinnaker/spinnaker/issues/5902)

If someone will provide configuration here, It will resolve my problem. What exactly should I write in `config`, and what exactly part should I put in  `gate-local.yml` to make it success.

That's all what I've got for now...
